addons/point_of_sale/models/pos_order.py

```diff
@@ -572,6 +572,52 @@ def remove_from_ui(self, server_ids):
         orders.sudo().unlink()
         return orders.ids
 
+    @api.model
+    def search_paid_order_ids(self, config_id, domain, limit, offset):
+        """Search for 'paid' orders that satisfy the given domain, limit and offset."""
+        default_domain = ['&', ('config_id', '=', config_id), '!', '|', ('state', '=', 'draft'), ('state', '=', 'cancelled')]
+        real_domain = AND([domain, default_domain])
+        ids = self.search(AND([domain, default_domain]), limit=limit, offset=offset).ids
+        totalCount = self.search_count(real_domain)
+        return {'ids': ids, 'totalCount': totalCount}
+
+    def _export_for_ui(self, order):
```

Comment:
Do we really need to create those functions "export_from_ui", can't we just return the result of read like it is done in the file 'models.js'

Reply:
If I'm not mistaken, we 'read' orm models such as pos.config, pos.category, product.product, etc, but not pos.order. The closest thing we got from fetching order is thru `get_table_draft_orders`, which in essence doing the same thing as the proposed method 'export_for_ui'.

This is what we have at the moment:
models.Order -> export_as_JSON -> create_from_ui -> pos.order

I am proposing this:
models.Order <- init_from_JSON <- export_for_ui <- pos.order

export_for_ui is just the parallel of export_as_JSON.

(It should have been easier if from the beginning, models.Order have the same signature as pos.order. But that is not the case.)

In essence, export_for_ui is just a version of get_table_draft_orders but I'm not sure if it's cleaner.

**As to why we represent models.Order for paid orders**
In the list of orders in the order management screen, I represented each order to be models.Order because the list can contain currently active orders. Representing them as models.Order is two-fold, 1) we have uniformity of data used in the order list, 2) we can recreate the receipt env which can be used to reprint the receipt.

Reply:
ok, you can leave it like that